### PR TITLE
[LW] Unify the NoOpLockWatchManager and its underlying cache

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- The NoOpLockWatchManager should still track updates from the timelock server in terms of its leadership.

**Implementation Description (bullets)**:
- We now create the NoOpLockWatchManager from an already-defined Event Cache, as opposed to creating it with a fresh (and different!) cache. Previously there were 2 event caches when using no-op, one which updates were read from and _another one_ where responses from timelock were processed.

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Nothing in particular.

**Concerns (what feedback would you like?)**:
- The implementation in TestTransactionManagerImpl is janky, though I didn't see a good way to remove it without refactoring the whole class to static factories :(

**Where should we start reviewing?**: NoOpLockWatchManager

**Priority (whenever / two weeks / yesterday)**: this week?

### CODE REVIEW

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public TestTransactionManagerImpl(
                 metricsManager,
                 createAssertKeyValue(keyValueService, lockService),
                 new LegacyTimelockService(timestampService, lockService, lockClient),
-                NoOpLockWatchManager.create(),
+                NoOpLockWatchManager.create(NoOpLockWatchEventCache.create()),
```
Comment (Jolyon-S) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java:106]: Could you not just factor this out to be a private final field that is instantiated there and then? i.e.
`private final LockWatchEventCache eventCache = NoOpLockWatchEventCache.create();`?

**Review by Jolyon-S — COMMENTED** [review]
Looks good - just a comment on how to maybe make the janky bit a bit less janky.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java:106]: In general, you can't reference even private fields before calling a supertype constructor (which is what is happening here).

Comment (Jolyon-S) [reply atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TestTransactionManagerImpl.java:106]: I didn't know that. Fair enough, then - I'll approve.

**Review by Jolyon-S — APPROVED** [review]

**Pull request merged**